zkabacus-crypto/src/revlock.rs

```diff
@@ -168,52 +214,78 @@ impl RevocationLockCommitment {
     pub(crate) fn verify_revocation_pair(
         &self,
         parameters: &merchant::Config,
-        revocation_secret: &RevocationSecret,
-        revocation_lock: &RevocationLock,
+        revocation_pair: &RevocationPair,
         revocation_lock_blinding_factor: &RevocationLockBlindingFactor,
     ) -> Verification {
-        let pair_is_valid = revocation_lock.verify(revocation_secret);
         let opening_is_valid = self.0.verify_opening(
             parameters.revocation_commitment_parameters(),
             revocation_lock_blinding_factor.0,
-            &Message::from(revocation_lock.to_scalar()),
+            &Message::from(revocation_pair.lock.to_scalar()),
         );
 
-        match (pair_is_valid, opening_is_valid) {
-            (Verification::Verified, true) => Verification::Verified,
-            _ => Verification::Failed,
+        if opening_is_valid {
+            Verification::Verified
+        } else {
+            Verification::Failed
         }
```

Comment:
Since we `impl From<bool> for Verification` you can do:

```rust
pub(crate) fn verify_revocation_pair(...) -> Verification {
    self.0.verify_opening(
      ...
    ).into()
}
```